### ISSUE TITLE
If the validator receives an actual `Decimal`, then the `toFixed` function stops working, breaking prisma.

### ISSUE THREAD

**Issue by emi2k01** [chrishoermann/zod-prisma-types#116]
Something like:

```ts
const fooInput = {
  val: new Decimal("10.5");
};

const parsedFooInput = FooCreateInputSchema.parse(input);

prisma.foo.create({
  data: fooInput,
})
```

The error given is:

```
Cannot read properties of undefined (reading 'constructor')
| TypeError: Cannot read properties of undefined (reading 'constructor')                                             
│    at P.toFixed (/home/emi2k01/src/work/gt/cs/qa/node_modules/decimal.js/decimal.js:2029:16)                       
│    at Object.toFixed (/home/emi2k01/src/work/gt/cs/qa/node_modules/zod/lib/types.js:2522:32)                       
│    at stringify (/home/emi2k01/src/work/gt/cs/qa/node_modules/@prisma/client/runtime/library.js:156:627)           ││                                                                                                                
│    at Object._toString (/home/emi2k01/src/work/gt/cs/qa/node_modules/@prisma/client/runtime/library.js:163:21)     ││                                                                                                                    
│    at Object.toString (/home/emi2k01/src/work/gt/cs/qa/node_modules/@prisma/client/runtime/library.js:163:68)      ││                                                                                                                    
│    at <anonymous> (/home/emi2k01/src/work/gt/cs/qa/node_modules/@prisma/client/runtime/library.js:155:658)         ││                                                                                                                    
│    at Array.map (<anonymous>)                                                                                                                                                                                                      
│    at Object.toString (/home/emi2k01/src/work/gt/cs/qa/node_modules/@prisma/client/runtime/library.js:155:649)     ││                                                                                                                    
│    at Object._toString (/home/emi2k01/src/work/gt/cs/qa/node_modules/@prisma/client/runtime/library.js:157:20)     ││                                                                                                                    
│    at Object.toString (/home/emi2k01/src/work/gt/cs/qa/node_modules/@prisma/client/runtime/library.js:163:68) {    ││                                                                                                                    
│  clientVersion: '4.11.0'                                                                                           
│} 
```

**Comment by chrishoermann** [comment]
@emi2k01 That's interesting. there is something wonky since prisma introduced the toFixed method in the `DecimalJsLike`. I think need to dig a bit deeper on this one 🤔

**Comment by chrishoermann** [comment]
This should be fixed no in `v3.0.0`
